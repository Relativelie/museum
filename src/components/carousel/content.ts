import slideOne from '../../assets/images/carousel/img.png';

export const carouselContent = [
    {
        id: 1,
        name: 'Cлайд 4',
        image: slideOne,
    },
    {
        id: 2,
        name: 'Cлайд 2',
        image: slideOne,
    },
    {
        id: 3,
        name: 'Cлайд 3',
        image: slideOne,
    },
    {
        id: 4,
        name: 'Cлайд 4',
        image: slideOne,
    },
];
