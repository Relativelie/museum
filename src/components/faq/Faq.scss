@import "../../variables";

.faq {
    display: grid;
    grid-template-columns: 352px minmax(50%, 988px);
    margin: 170px 100px 100px 100px;
}

.faq_title {

    & h3 {
        margin: 0;
        font-family: $fontFamily;
        font-size: 80px;
        color: $fontColor;
    }
}

details[open] {
    border-top: 1px solid rgba(230, 210, 210) !important;
}

.faqContent_elem {
    border-top: 1px solid rgba(230, 210, 210, 0.2);

    &:last-child {
        border-bottom: 1px solid rgba(230, 210, 210, 0.2);
    }

    & .faqContent_question {
        padding: 30px 0;
        font-size: 24px;
        list-style-type: none;
        font-family: $fontFamily;
        color: $fontColor;
        cursor: pointer;
    }

    & .faqContent_answer {
        margin-bottom: 30px;
        font-family: $fontFamily;
        font-size: 16px;
        color: $fontColor;
    }
}