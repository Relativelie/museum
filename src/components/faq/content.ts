export const faqContent = [
    {
        id: 1,
        question: 'Как добраться до музея?',
        answer: 'В Национальный музей Республики Татарстан можно добраться общественным транспортом:1) «Станция метро «Кремлевская», а затем по направлению к Кремлю пройдя 5 минут пешком;',
    },
    {
        id: 2,
        question: 'Есть ли поблизости удобная парковка?',
        answer: 'some text',
    },
    {
        id: 3,
        question: 'Можно ли воспользоваться аудиогидом?',
        answer: 'some text',
    },
];
