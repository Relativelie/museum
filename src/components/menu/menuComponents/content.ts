export const menuContent = [
    {
        name: 'Пункт 1',
        id: 1001,
        content: [
            { id: 10011, text: 'Подпункт меню1', url: '/someUrl' },
            { id: 10012, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10013, text: 'Подпункт меню1', url: '/someUrl' },
            { id: 10014, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10015, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 2',
        id: 1002,
        content: [
            { id: 10021, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10022, text: 'Подпункт меню2', url: '/someUrl' },
            { id: 10023, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10024, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10025, text: 'Подпункт меню2', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 3',
        id: 1003,
        content: [
            { id: 10031, text: 'Подпункт меню3', url: '/someUrl' },
            { id: 10032, text: 'Подпункт меню3', url: '/someUrl' },
            { id: 10033, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10034, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10035, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 4',
        id: 1004,
        content: [
            { id: 10041, text: 'Подпункт меню4', url: '/someUrl' },
            { id: 10042, text: 'Подпункт меню4', url: '/someUrl' },
            { id: 10043, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10044, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10045, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 5',
        id: 1005,
        content: [
            { id: 10051, text: 'Подпункт меню5', url: '/someUrl' },
            { id: 10052, text: 'Подпункт меню5', url: '/someUrl' },
            { id: 10053, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10054, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
];
