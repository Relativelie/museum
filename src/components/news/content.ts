import imageOne from '../../assets/images/news/newsOne.png';
import imageTwo from '../../assets/images/news/newsTwo.png';
import imageThree from '../../assets/images/news/newsThree.png';
import imageFour from '../../assets/images/news/newsFour.png';
import imageFive from '../../assets/images/news/newsFive.png';
import imageSix from '../../assets/images/news/newsSix.png';

export const content = [
    {
        id: 4001,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageOne,
    },
    {
        id: 4002,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageTwo,
    },
    {
        id: 4003,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageThree,
    },
    {
        id: 4004,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageFour,
    },
    {
        id: 4005,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageFive,
    },
    {
        id: 4006,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageSix,
    },
];
