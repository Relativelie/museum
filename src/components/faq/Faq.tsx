import { FC } from 'react';
import { faqContent } from './content';
import './Faq.scss';

export const Faq: FC = () => {
    return (
        <div className="faq">
            <div className="faq_title">
                <h3>FAQ</h3>
            </div>

            <div className="faqContent">
                {faqContent.map(((elem) => {
                    const { id, question, answer } = elem;
                    return (
                        <details className="faqContent_elem" key={`faq-${id}`}>
                            <summary className="faqContent_question">{question}</summary>
                            <p className="faqContent_answer">{answer}</p>
                        </details>
                    );
                }))}
            </div>
        </div>
    );
};
