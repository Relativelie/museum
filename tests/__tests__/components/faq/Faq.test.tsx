import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act, create } from 'react-test-renderer';
import { screen } from '@testing-library/dom';
import { Faq } from '../../../../src/components/faq/Faq';
import '@testing-library/jest-dom';

import { faqContent } from '../../../../src/components/faq/content';

describe('FAQ component', () => {
    test('snapshot - FAQ component', () => {
        let root: any;
        act(() => {
            root = create(<Faq />);
        });
        expect(root.toJSON()).toMatchSnapshot();
    });

    test('snapshot - FAQ component', () => {
        render((<Faq />));
        expect(document.querySelector('[open]')).toBeNull();
        userEvent.click(screen.getByText(faqContent[0].question));
        expect(document.querySelector('.faqContent_elem')).toHaveAttribute('open');
    });
});
