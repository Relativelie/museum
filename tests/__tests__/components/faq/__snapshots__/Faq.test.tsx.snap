// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ component snapshot - FAQ component 1`] = `
<div
  className="faq"
>
  <h3
    className="faq_title"
  >
    FAQ
  </h3>
  <div
    className="faqContent"
  >
    <details
      className="faqContent_elem"
    >
      <summary
        className="faqContent_question"
      >
        Как добраться до музея?
      </summary>
      <p
        className="faqContent_answer"
      >
        В Национальный музей Республики Татарстан можно добраться общественным транспортом:1) «Станция метро «Кремлевская», а затем по направлению к Кремлю пройдя 5 минут пешком;
      </p>
    </details>
    <details
      className="faqContent_elem"
    >
      <summary
        className="faqContent_question"
      >
        Есть ли поблизости удобная парковка?
      </summary>
      <p
        className="faqContent_answer"
      >
        some text
      </p>
    </details>
    <details
      className="faqContent_elem"
    >
      <summary
        className="faqContent_question"
      >
        Можно ли воспользоваться аудиогидом?
      </summary>
      <p
        className="faqContent_answer"
      >
        some text
      </p>
    </details>
  </div>
</div>
`;
